### ISSUE TITLE
question

### ISSUE THREAD

**Issue by changshh** [mephisto121/DeepGenAntiOxidantPeptide#1]
mxlen=19
train_pading = generate_data.padding(train_encode,mxlen=mxlen)
test_padding = generate_data.padding(test_encode,mxlen=mxlen)
print("Train data shape:", train_pading.shape)
print("Test data shape:", test_padding.shape)
print("Train data example:", train_pading[0])
print("Test data example:", test_padding[0])

x_train, y_train = generate_data.split_for_data(train_pading)
x_test, y_test = generate_data.split_for_data(test_padding)

# In[7]:

x_train[1], y_train[1]

# In[ ]:
print("Shape of x_train:", x_train.shape)
print("Shape of y_train:", y_train.shape)
print("Length of a sample in x_train:", len(x_train[0]))

已经设置为mxlen=19，为什么后面打印出来的是18

Shape of x_train: (122716, 18)
Shape of y_train: (122716, 18)
Length of a sample in x_train: 18

**Comment by changshh** [comment]
要设置mxlen=20，解决问题

**Comment by mephisto121** [comment]
Hi. I think the main problem should align on the tensorflow version. In the code, I used tensorflow 2.7. However, to do some checks and create metrics, later I had to use a newer version, and you can see the difference in the Conv1D classification model where they moved the Adam optimizer to the legacy. please try the code with tensorflow 2.7 and see if you get the same issue or not.
